client/rpc_sender.go

```diff
@@ -83,6 +84,13 @@ func newRPCSender(server string, context *base.Context, retryOpts retry.Options,
 // and been executed successfully. We retry here to eventually get through with
 // the same client command ID and be given the cached response.
 func (s *rpcSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.Error) {
+	timeout := 5 * s.retryOpts.MaxBackoff
```

Comment:
Are you sure the time is being spent in `Get{Client,Server}TLSConfig`? Those methods are called synchronously in `rpc.NewClient` and `server.NewServer`, so it seems they are already guaranteed to have returned before any timeouts are involved.